### ISSUE TITLE
Gifhive

### ISSUE THREAD

**Issue by iSasFTW** [status-im/hackathon#10]
### Gifhive - Decentralized GIF sharing

Type: DApp
Github Repo: https://github.com/iSasFTW/swarm-gif-share
Description:

A small dapp that lets the user post a GIF with a caption, and share it with other users quickly over Swarm. (+ Szabo tipping functionality)

Demo video: https://youtu.be/fl3yzHerfsY

Team Members:

Preferred Name: Saska
Github Handle:  @isasftw
Slack username: @isasftw
Twitter handle: @isasftw

**Comment by commiteth** [comment]
![Contract deploying](https://commiteth.com/img/deploying_contract.png)

**Comment by iSasFTW** [comment]
The UI with styling is now done. Next up upload, search and viewing. Viewing and upload I have a clear plan for, and therefore they should be quite straightforward to implement. Search is still a bit unclear. It is a necessary function, but it seems like making a decentralized database that can be searched is turning out to be harder than I thought.

**Comment by iSasFTW** [comment]
Uploading will be replaced by a set of pre-set images you can upload, due to a limitation with Status. The javascript code for uploading and converting to Base 64 will still be there, just commented out.

**Comment by iSasFTW** [comment]
Uploading to swarm is now done. At the moment it can only upload the preset images (just a few random gifs I downloaded) but the logic is there for uploading custom ones too. Next up I'll need to create viewing, and the hardest part, search. Not exactly sure how Weipoint works, but if they are using a smart contract for storing data, I might have to take a note from their books.

**Comment by iSasFTW** [comment]
The bugs regarding uploading are now fixed. I had a weird problem with an error I didn't recognize, but turns out that all I had to do was use regular function notation instead of the new ES6 arrow shorthand notation. Viewing is now done, too. It uses a query string in the url, that is a Swarm hash. The page gets that hash and loads the JSON data, and displays it to the user!

There are a few graphical bugs with the layout on the upload page, but I'll fix those in no time.
Tipping should be fairly simple. I'll just create a transaction with a specific ETH amount entered by the user (most likely in szabos) and send it to the poster of the image (address recorded to Swarm).

Still not sure on how I'll tackle search. Most likely I'll try some sort of approach with a Smart Contract, hopefully not too complex.

Hopefully I'll have time to add some Status Commands too, like /search (query) and /upload. 

Next I'll probably do some research on Solidity, since I've never written a smart contract before. I'll also have to try to get my Ethereum Wallet to work, to actually deploy the contract to Ropsten. Tipping will be a bit simpler though, so I might start with that instead.

**Comment by iSasFTW** [comment]
Tipping is now done. Now I have literally everything down, except for Status commands, and search.
Today I experimented with Solidity for the first time ever. I wrote my first, most likely terrible, contract. I can't find any existing examples for contracts that fit my needs for this dapp, which is a problem, since I only have so much time to get search working. Status commands at this point are not a priority, since without search, there's really nothing special to make a command of.  I'm concerned I might not have enough time to finish with making a Smart contract, inserting data into it, and trying to get back that data so I can search it. If I don't finish search in time, the dapp will just feel... Incomplete, and that is definitely not something I want.
Hopefully two days will be enough!
